appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AmountChooserPresenter.kt

```diff
@@ -29,6 +29,27 @@ class AmountChooserPresenter(private val appIconUrl: String, private val balance
         view?.sendKin(amount)
     }
 
+    override fun afterTextChanged(editable: Editable?) {
+        editable?.let {
+            if (editable.length >= 2) {
+                if (editable[0] == '0') {
+                    editable.delete(0, 1)
+                }
+            }
+            amount = 0
+            if (editable.isNotEmpty()) {
+                amount = try {
+                    editable.toString().toInt()
+                } catch (e: NumberFormatException) {
+                    editable.delete(editable.length - 2, editable.length - 1)
+                    editable.toString().toInt()
+                }
+            }
+            view?.setSendEnable(amount in 1..balance || balance == 0)
```

Comment:
why is send enabled when balance == 0 ?

Reply:
if balance is not recieved the default is zero and there is not limit how much to enter
very edge case